plugins/scanners/scancode/src/main/kotlin/ScanCodeResultParser.kt

```diff
@@ -52,137 +89,90 @@ private data class LicenseMatch(
-                    path = file["path"].textValue(),
-                    startLine = licenseMatch.startLine,
-                    endLine = licenseMatch.endLine
+                    path = file.path,
+                    startLine = license.startLine,
+                    endLine = license.endLine
                 ),
-                score = licenseMatch.score
+                score = license.score
             )
         }
-    }
-
-    return associateLicensesWithExceptions(licenseFindings)
-}
-
-/**
- * Get the SPDX license id (or a fallback) for a license finding.
- */
-private fun getSpdxLicenseId(license: JsonNode): String {
-    // There is a bug in ScanCode 3.0.2 that returns an empty string instead of null for licenses unknown to SPDX.
-    val idFromSpdxKey = license["spdx_license_key"].textValueOrEmpty().toSpdxId(allowPlusSuffix = true)
 
-    // For regular SPDX IDs, return early here.
-    if (idFromSpdxKey.isNotEmpty()) return idFromSpdxKey
-
-    // At this point the ID is either empty or a non-ScanCode SPDX LicenseRef, so fall back to building an ID based
-    // on the ScanCode-specific "key".
-    val idFromKey = license["key"].textValue().toSpdxId(allowPlusSuffix = true)
-
-    return "$LICENSE_REF_PREFIX_SCAN_CODE$idFromKey"
-}
-
-/**
- * Get the copyright findings from the given [result].
- */
-private fun getCopyrightFindings(result: JsonNode): Set<CopyrightFinding> {
-    val copyrightFindings = mutableSetOf<CopyrightFinding>()
-
-    val header = result["headers"].single()
-    val outputFormatVersion = header["output_format_version"]?.textValue()?.let { Semver(it) }
-    val copyrightKeyName = if (outputFormatVersion == null || outputFormatVersion < Semver("2.0.0")) {
-        "value"
-    } else {
-        "copyright"
-    }
-
-    val files = result["files"]?.asSequence().orEmpty()
-
-    files.flatMapTo(copyrightFindings) { file ->
-        val path = file["path"].textValue()
-
-        val copyrights = file["copyrights"]?.asSequence().orEmpty()
-        copyrights.map { copyright ->
+        file.copyrights.mapTo(copyrightFindings) { copyright ->
             CopyrightFinding(
-                statement = copyright[copyrightKeyName].textValue(),
+                statement = copyright.statement,
                 location = TextLocation(
-                    path = path,
-                    startLine = copyright["start_line"].intValue(),
-                    endLine = copyright["end_line"].intValue()
+                    path = file.path,
+                    startLine = copyright.startLine,
+                    endLine = copyright.endLine
                 )
             )
         }
     }
 
-    return copyrightFindings
+    return ScanSummary(
+        startTime = TIMESTAMP_FORMATTER.parse(header.startTimestamp).query(Instant::from),
+        endTime = TIMESTAMP_FORMATTER.parse(header.endTimestamp).query(Instant::from),
+        licenseFindings = associateLicensesWithExceptions(licenseFindings),
```

Comment:
Maybe `fun parseTimeStamp(String): Instant` could be extracted.

Reply:
Right, but it was inlined also before, and introducing a function doesn't really pay off here, IMO.